tf2onnx/shape_inference.py

```diff
@@ -213,6 +210,40 @@ def infer_input_shapes(g, node):
 
 
 def infer_output_shapes_with_partial_inputs(g, node):
+    if node.type == "ConcatV2":
+        axis_node = node.inputs[-1]
+        if not axis_node.is_const():
+            return False
+
+        axis = axis_node.get_tensor_value()
+        data_inputs = node.input[:-1]
+        input_shapes = [g.get_shape(node) for node in data_inputs]
+        # find the best shape info, concat node requires all dims equal except the concat dim
+        new_shape = input_shapes[0]
+        for tmp in input_shapes:
+            if tmp is not None:
+                if new_shape is None:
+                    new_shape = tmp
+                elif tmp.count(-1) < new_shape.count(-1):

```

Comment:
if i understand correctly, what we really want to do is below, right?
```
if new_shape[i] == -1
    new_shape[i] = tmp[i]
```

comparing count of -1 does not seem to be right.